server/src/chat/chat.service.ts

```diff
@@ -166,128 +209,142 @@ export class ChatService {
     }
 
+    // @TODO: outsource notifying other chat members, that someone new joined -> gateway / gateway state helper
     async joinChat(user: User, chatId: string) {
         try {
-            let chatRoom = await this.prisma.chatRoom.findFirst({
-                where: { id: chatId, ...WHERE_user_id(user.id) },
-                // include: {
-                //     users: SELECT_user,
-                //     createdBy: SELECT_user,
-                // },
+            let chatGroup = await this.prisma.chatGroup.findFirst({
+                where: { id: chatId, ...WHERE_member(user.id) },
                 ...SELECT_chat_preview,
             });
-            if (chatRoom) return { chatRoom, successMessage: 'You already joined the chat' };
+            if (chatGroup) return { chatRoom: chatGroup, successMessage: 'You already joined the chat' };
 
-            chatRoom = await this.prisma.chatRoom.update({
+            chatGroup = await this.prisma.chatGroup.update({
                 where: { id: chatId },
-                data: { users: { connect: { id: user.id } } },
-                // include: {
-                //     users: SELECT_user,
-                //     createdBy: SELECT_user,
-                // },
+                data: {
+                    members: { create: { user: { connect: { id: user.id } } } },
+                },
                 ...SELECT_chat_preview,
             });
-            if (!chatRoom) throw new NotFoundException();
+            if (!chatGroup) throw new NotFoundException();
+
+            const onlineUser = this.getUserOnline(user.id, 'userId');
+            if (onlineUser) {
+                this.addUserOnlineToChat(onlineUser, chatId, true);
+                onlineUser.client.broadcast.to(chatId).emit(SocketEvents.SERVER__USER_JOINED_CHAT, {
+                    chatId,
+                    user: { id: user.id, username: user.username },
+                });
+            }
 
             return {
-                chatRoom,
-                successMessage: `Successfully joined chat '${chatRoom.title || chatRoom.id}'`,
+                chatRoom: chatGroup,
+                successMessage: `Successfully joined chat '${chatGroup.title || chatGroup.id}'`,
             };
         } catch (err) {
             throw new NotFoundException();
         }
     }
-    async leaveChat(user: User, chatRoomId: string) {
-        try {
-            let chatRoom = await this.prisma.chatRoom.findFirst({
-                where: { id: chatRoomId, ...WHERE_user_id(user.id) },
-            });
-            if (!chatRoom) return { successMessage: 'You already left the chat' };
+    async leaveChat(user: User, chatId: string) {
+        // let membership = await this.prisma.membership.findFirst({
+        //     where: {
+        //         chatGroupId: chatId,
+        //         userId: user.id,
+        //     },
+        // });
+        // if (!membership) return { successMessage: 'You already left the chat' };
```

Comment:
Lets get rid of these comments, if it works, it works, otherwise we need to investigate/refactor anyway.